torchvision/prototype/models/_api.py

```diff
@@ -101,6 +101,11 @@ def get_weight(fn: Callable, weight_name: str) -> Weights:
         # TODO: Replace ann.__args__ with typing.get_args(ann) after python >= 3.8
         for t in ann.__args__:  # type: ignore[union-attr]
             if isinstance(t, type) and issubclass(t, Weights):
+                # ensure the name exists. handles builders with multiple types of weights like in quantization
+                try:
+                    t.from_str(weight_name)
+                except ValueError:
+                    continue
```

Comment:
Shouldn't we at least print a warning related to the exception rather than just ignoring it?

Reply:
Ignoring is probably the right thing to do here. We try to match the provided string of the user with any known weight enums and if all fail, on the next couple of lines we throw an exception:
https://github.com/pytorch/vision/blob/015ee04a54908fc3ba9df65587d682f3d56268cf/torchvision/prototype/models/_api.py#L112-L115

Throwing a warning for every enum trying to match will only cause confusion.